components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/LoginEventHookHandler.java

```diff
@@ -0,0 +1,224 @@
+                .getLoginEventPayloadBuilder(EVENT_SCHEMA_TYPE_WSO2);
+        EventAttribute loginEventPublisherConfig = getLoginEventPublisherConfigForTenant(
+                eventData.getAuthenticationContext().getLoginTenantDomain(), event.getEventName());
+        EventPayload eventPayload;
+        String eventUri;
+
+        if (IdentityEventConstants.EventName.AUTHENTICATION_SUCCESS.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationSuccessEvent(eventData);
+            eventUri = eventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_SUCCESS_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+            eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        } else if (IdentityEventConstants.EventName.AUTHENTICATION_STEP_FAILURE.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationFailedEvent(eventData);
+            eventUri = eventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_FAILED_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+            eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        }
+    }
+
+    private EventAttribute getLoginEventPublisherConfigForTenant(String tenantDomain, String eventName) {
+
+        if (StringUtils.isEmpty(tenantDomain) || MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            return new EventAttribute();
+        }
+
+        try {
+            Condition condition = createPublisherConfigFilterCondition();
+            Resources publisherConfigResource = EventHookHandlerDataHolder.getInstance().getConfigurationManager()
+                    .getTenantResources(tenantDomain, condition);
+
+            return extractEventAttribute(publisherConfigResource, eventName);
+        } catch (ConfigurationManagementException | IdentityEventException e) {
+            log.debug("Error while retrieving event publisher configuration for tenant.", e);
+        }
+
+        return new EventAttribute();
+    }
+
+    private EventAttribute extractEventAttribute(Resources publisherConfigResource, String eventName) throws IdentityEventException {
+
+        if (CollectionUtils.isNotEmpty(publisherConfigResource.getResources()) &&
+                publisherConfigResource.getResources().get(0) != null &&
```

Comment:
What's the reason to check always the 0th index object ?

Reply:
we are retrieving the tenanted resources through a filter condition and the result got from is an array list. That's why checking for the 0th index value.